sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -61,10 +66,59 @@ public abstract class AbstractCallGraphAlgorithm implements CallGraphAlgorithm {
 
   @NonNull protected final View view;
   @NonNull protected final TypeHierarchy typeHierarchy;
+  @NonNull protected final Table<String, String, SootMethod> preanalysis;
 
   protected AbstractCallGraphAlgorithm(@NonNull View view) {
     this.view = view;
     this.typeHierarchy = view.getTypeHierarchy();
+    this.preanalysis = getMethodsWithPolymorphicAnnotation();
+  }
+
+  /**
+   * Collects all method signatures from the current view that are annotated with {@code
+   * java.lang.invoke.MethodHandle$PolymorphicSignature}.
+   *
+   * @return a list of method signatures annotated with {@code @PolymorphicSignature}
+   */
+  protected Table<String, String, SootMethod> getMethodsWithPolymorphicAnnotation() {
+    Table<String, String, SootMethod> polymorphicMethods = HashBasedTable.create();
+    ClassType polymorphicAnnotationType =
+        view.getIdentifierFactory()
+            .getClassType("java.lang.invoke.MethodHandle$PolymorphicSignature");
+    Map<ClassType, List<JavaSootMethod>> polymorphicTypeAndMethods = new HashMap<>();
```

Comment:
we dont need to save this.
Small Info: In this case a ArrayListMultimap would be easier to manage instead of a Map containing a List